File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/presenter/PlayerPresenter.java
package com.dbottillo.mtgsearchfree.presenter;

import com.dbottillo.mtgsearchfree.model.Player;
import com.dbottillo.mtgsearchfree.view.PlayersView;

import java.util.List;

public interface PlayerPresenter extends BasicPresenter {

    void init(PlayersView view);

    void loadPlayers();

    void addPlayer();

    void editPlayer(Player player);

    void editPlayers(List<Player> players);

    void removePlayer(Player player);

}

File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/presenter/PlayerPresenter.kt
package com.dbottillo.mtgsearchfree.presenter

import com.dbottillo.mtgsearchfree.model.Player
import com.dbottillo.mtgsearchfree.view.PlayersView

interface PlayerPresenter : BasicPresenter {

    fun init(view: PlayersView)

    fun loadPlayers()

    fun addPlayer()

    fun editPlayer(player: Player)

    fun editPlayers(players: List<Player>)

    fun removePlayer(player: Player)

}